src/potentiometer-element.ts

```diff
@@ -0,0 +1,269 @@
+            />
+            <ellipse
+              ry="43.629311"
+              rx="43.739475"
+              cy="-139.91585"
+              cx="-120.38097"
+              style="opacity:1;fill:#c3c2c3;fill-opacity:1;stroke:none;stroke-width:1.77517259;stroke-linecap:butt;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
+            />
+            <rect
+              id="${this.id}"
+              y="-180.5804"
+              x="-120.168198"
+              height="20.26368"
+              width="2.7833986"
+            />
+          </g>
+        </g>
+      </svg>
+    `;
+  }
+
+  private down(event: MouseEvent) {
+    if (event.button === 0 || window.navigator.maxTouchPoints) {
+      this.pressed = true;
+      this.updatePotentiometerPosition(event);
+    }
+  }
+
+  private move(event: MouseEvent) {
+    const { pressed } = this;
+    if (pressed) {
+      this.rotateHandler(event);
+    }
+  }
+
+  private up() {
+    this.pressed = false;
+  }
+
+  private updatePotentiometerPosition(event: MouseEvent | TouchEvent) {
+    event.stopPropagation();
+    event.preventDefault();
+
+    const potentiometerRect = this.getBoundingClientRect();
+
+    this.center = {
+      x: window.scrollX + potentiometerRect.left + potentiometerRect.width / 2,
+      y: window.scrollY + potentiometerRect.top + potentiometerRect.height / 2
+    };
+  }
+
+  private rotateHandler(event: MouseEvent | TouchEvent) {
+    event.stopPropagation();
+    event.preventDefault();
+
+    const isTouch = event.type === 'touchmove';
+    const pageX = isTouch ? (event as TouchEvent).touches[0].pageX : (event as MouseEvent).pageX;
+    const pageY = isTouch ? (event as TouchEvent).touches[0].pageY : (event as MouseEvent).pageY;
+    const x = this.center.x - pageX;
+    const y = this.center.y - pageY;
+    let deg = Math.round((Math.atan2(y, x) * 180) / Math.PI);
+
+    if (deg < 0) {
+      deg += 360;
+    }
+
+    deg -= 90;
+
+    if (x > 0 && y <= 0) {
+      deg -= 360;
+    }
+
+    deg = this.clamp(this.startDegree, this.endDegree, deg);
+    const percent = this.percentFromMinMax(deg, this.startDegree, this.endDegree);
+    const value = this.mapToMinMax(percent, this.min, this.max);
+
+    this.updateValue(value);
+  }
+
+  private updatePotentiometerPointer(value: number) {
+    const percent = this.percentFromMinMax(value, this.min, this.max);
+    const deg = Math.round((this.endDegree - this.startDegree) * percent + this.startDegree);
+    this.style.setProperty('--knob-angle', `${deg}deg`);
+  }
+
+  private updateValue(value: number) {
+    const clamped = this.clamp(this.min, this.max, value);
+    const updated = Math.round(clamped / this.step) * this.step;
+    const rounded = Math.round(updated * 100) / 100;
+    this.updatePotentiometerPointer(rounded);
+    this.dispatchEvent(new CustomEvent('potentiometer-value-changed', { detail: rounded }));
```

Comment:
How about dispatching an [`InputEvent`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event)? 
This will make the behavior of the element more in line with a standard  `<input>` element

Reply:
InputEvent it will be